builtin/logical/consul/secret_token.go

```diff
@@ -37,7 +39,12 @@ func secretTokenRevoke(
 		return logical.ErrorResponse(err.Error()), nil
 	}
 
-	_, err = c.ACL().Destroy(d.Get("token").(string), nil)
+	tokenRaw, ok := req.Secret.InternalData["token"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing internal data: token")
+	}
+
+	_, err = c.ACL().Destroy(tokenRaw.(string), nil)
 	if err != nil {
```

Comment:
Should this then return a real error, not a logical.ErrorResponse, so the expiration manager sees that it wasn't successful?